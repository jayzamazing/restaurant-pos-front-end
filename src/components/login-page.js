import React from 'react';
import {connect} from 'react-redux';
import {Link, Redirect} from 'react-router-dom';
import LoginForm from './login-form';

export function LoginPage(props) {
    // If we are logged in redirect straight to the user's dashboard
    if (props.loggedIn) {
        return <Redirect to="/boards" />;
    }

    return (
        <div className="login container">
            <div className="row">
              <div className="col-centered login-area">
                <h2>Log in to Trello</h2>
                <LoginForm />
                <p>Don't have an account? <Link to="/register">Create a Trello Account.</Link></p>
              </div>
            </div>
        </div>
    );
}

const mapStateToProps = state => ({
    loggedIn: state.auth.currentUser !== null
});

export default connect(mapStateToProps)(LoginPage);
