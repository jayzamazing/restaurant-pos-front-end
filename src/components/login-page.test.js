'use strict';
import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import {LoginPage} from './login-page';
import {Redirect} from 'react-router-dom';

describe('home-page', () => {
  it('should render page', () => {
    shallow(<LoginPage />);
  });
  it('snapshot', () => {
    const wrapper = shallow(<LoginPage />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should redirect when user is logged in', () => {
    const wrapper = shallow(<LoginPage loggedIn="true"/>);
    expect(wrapper.find(Redirect)).toHaveLength(1);
  });
  it('redirect snapshow', () => {
    const wrapper = shallow(<LoginPage loggedIn="true"/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
