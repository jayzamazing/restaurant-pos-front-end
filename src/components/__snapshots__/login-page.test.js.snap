// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home-page redirect snapshow 1`] = `
<Redirect
  push={false}
  to="/tables"
/>
`;

exports[`home-page snapshot 1`] = `
<div
  className="login container"
>
  <div
    className="row"
  >
    <div
      className="col-centered login-area"
    >
      <h3>
        Welcome to Restaurant POS.
      </h3>
      <p>
        Please use the following credentials in order to login.
      </p>
      <Connect(ReduxForm) />
    </div>
  </div>
</div>
`;
